src/modules/ADAR/components/RouteAssets/SwapDialog.vue

```diff
@@ -1,22 +1,67 @@
 <template>
   <div class="swap-dialog">
     <dialog-base :visible.sync="isVisible" :title="'Swap'" custom-class="dialog__swap">
-      <Swap />
+      <Swap ref="swap" />
     </dialog-base>
   </div>
 </template>
 
 <script lang="ts">
-import { Component, Mixins } from 'vue-property-decorator';
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator';
 import { mixins, components } from '@soramitsu/soraneo-wallet-web';
 import Swap from '@/views/Swap.vue';
+import type { PresetSwapData } from '@/store/routeAssets/types';
+import { FPNumber, NetworkFeesObject, Operation } from '@sora-substrate/util/build';
+import { state } from '@/store/decorators';
+import { XOR } from '@sora-substrate/util/build/assets/consts';
 @Component({
   components: {
     DialogBase: components.DialogBase,
     Swap,
   },
 })
-export default class SwapDialog extends Mixins(mixins.TransactionMixin, mixins.DialogMixin) {}
+export default class SwapDialog extends Mixins(mixins.TransactionMixin, mixins.DialogMixin) {
+  @state.wallet.settings.networkFees private networkFees!: NetworkFeesObject;
+
+  @Prop({ default: 0 }) presetSwapData!: PresetSwapData;
+
+  get presetDataValueTo() {
+    return this.presetSwapData?.valueTo;
+  }
+
+  roundNumber(num) {
+    return Math.ceil((num + Number.EPSILON) * 100) / 100;
+  }
+
+  get networkSwapFee(): number {
+    return FPNumber.fromCodecValue(this.networkFees[Operation.Swap]).toNumber() * 2;
```

Comment:
what is it?

Reply:
This is preset value in swap dialog, if you don't have enough XOR for tx taxes.
0,7 XOR - tax for transfer, and 0,7 XOR - tax for swap operation to required XOR amount